core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1555,22 +1555,42 @@ class WorkspaceService(protected val userInfo: UserInfo,
 
   def createSubmission(workspaceName: WorkspaceName, submissionRequest: SubmissionRequest): Future[SubmissionReport] = {
     for {
-      (workspaceContext, submissionParameters, workflowFailureMode, header) <- prepareSubmission(workspaceName, submissionRequest)
-      submission <- saveSubmission(workspaceContext, submissionRequest, submissionParameters, workflowFailureMode, header)
+      (workspaceContext, submissionId, submissionParameters, workflowFailureMode, header, outputPath) <- prepareSubmission(workspaceName, submissionRequest)
+      submission <- saveSubmission(workspaceContext, submissionId, submissionRequest, outputPath, submissionParameters, workflowFailureMode, header)
     } yield {
       SubmissionReport(submissionRequest, submission.submissionId, submission.submissionDate, userInfo.userEmail.value, submission.status, header, submissionParameters.filter(_.inputResolutions.forall(_.error.isEmpty)))
     }
   }
 
   private def prepareSubmission(workspaceName: WorkspaceName, submissionRequest: SubmissionRequest):
-  Future[(Workspace, Stream[SubmissionValidationEntityInputs], Option[WorkflowFailureMode], SubmissionValidationHeader)] = {
+  Future[(Workspace, UUID, Stream[SubmissionValidationEntityInputs], Option[WorkflowFailureMode], SubmissionValidationHeader, String)] = {
+
+    val submissionId: UUID = UUID.randomUUID()
+
     for {
+      workspaceContext <- getWorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.write)
+
       _ <- requireComputePermission(workspaceName)
 
+      outputPath = submissionRequest.outputPath match {
+        case None => s"gs://${workspaceContext.bucketName}/${submissionId.toString}"
+        case Some(path) => {
+          val pathFormatted = path.stripSuffix("/").toLowerCase
+
+          parseGcsPath(pathFormatted) match {
+            case Left(error) => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"The specified outputPath was invalid. ${error.value}"))
+            case Right(GcsPath(GcsBucketName(bucketName), GcsObjectName(objectName, _))) =>
+              if(!bucketName.equalsIgnoreCase(s"${workspaceContext.bucketName}")) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"The specified outputPath must be within the workspace bucket gs://${workspaceContext.bucketName}"))
+              if(objectName.nonEmpty && !objectName.matches(gcsObjectNameRegex.regex)) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, """The specified outputPath was invalid. Path within bucket must be between 1 and 1024 characters and only contain alphanumeric characters or characters in the following list: !@$%^&(){}|<>/,.=_-."""))
+
+              s"${pathFormatted}/${submissionId.toString}"
+          }
+        }
+      }
+
       // getWorkflowFailureMode early because it does validation and better to error early
       workflowFailureMode <- getWorkflowFailureMode(submissionRequest)
```

Comment:
This comment seems a little out of date now. Could this still be the first thing we do? If not, I think what you've added is also in the spirit of validation, but that validation also now needs `getWorkspaceContextAndPermissions`. The comment might need to moved, be rephrased, or removed altogether?

Reply:
I've moved it back up near the top of the `for` in 293e12f41f7c73e9021b41fe834596d91fcf58bd